### PR TITLE
Change bibliography to footbibliography

### LINKED ISSUE #7: codecov badge shows 0%
## Expected behavior ##

Coverage is properly accounted for and displayed.

## Actual behavior ##

Codecov badge shows 0% and there's no coverage.

### LINKED ISSUE #12: bibtex bibliography should be footbibliography
## Expected behavior ##

If you want a local bibliography in similarity.py, the correct directives to use are actually `footbibliography` and `footcite`. `cite` and `bibliography` are to be used for global bibliographies, like a separate references page. 

Making this change allowed me to build docs as below:

Screenshot 2023-08-24 at 9 48 48 am

### PR DESCRIPTION
**Pull request by lilyminium**
Fixes #12 

Changes made in this Pull Request:

 - Changes bibliography directive to footbibliography

PR Checklist
------------
 - [ ] Tests?
 - [ ] Docs?
 - [ ] CHANGELOG updated?
 - [ ] Issue raised/referenced?

### CODE REVIEW

**Comment by lilyminium** [comment]
This is more to show what I mean and the content that let me render what's in #12 -- it may make more sense to just have a global bibliography instead.

**Comment by orbeckst** [comment]
The RTD build https://readthedocs.org/projects/mdaencore/builds/21702726/ fails with
```
ModuleNotFoundError: No module named 'mdaencore.cutils'
```

**Comment by orbeckst** [comment]
Building locally `make html` and having installed with `pip install -e .`

1. `mdaencore.data.files` is not found:
   ```
   Failed to import mdaencore.data.files.
   Possible hints:
   * ModuleNotFoundError: No module named 'mdaencore.data.files'
   * AttributeError: module 'mdaencore.data' has no attribute 'files'
   WARNING: [autosummary] failed to import mdaencore.data.files.
   Possible hints:
   * ModuleNotFoundError: No module named 'mdaencore.data.files'
   * AttributeError: module 'mdaencore.data' has no attribute 'files'
   ```

2. Some footnote issues
   ```
   ~/MDAnalysis/mdaencore/docs/source/autosummary/mdaencore.rst:38:<autosummary>:1: ERROR: Too many autonumbered footnote references: only 0 corresponding footnotes available.
   ~/MDAnalysis/mdaencore/docs/source/autosummary/mdaencore.rst:38:<autosummary>:1: ERROR: Unknown target name: "footcite-lindorfflarsen2009".
   ```

On the plus side, the RTD build is working again.

File: docs/requirements.yaml [hunk 1]
```diff
@@ -8,11 +8,12 @@ dependencies:
     - pip
 
     - sphinxcontrib-bibtex=2.5.0
+    - mdanalysis-sphinx-theme
```
Comment (orbeckst) [inline docs/requirements.yaml:11]: I forgot to merge main into the branch before editing and had independently moved danalysis-sphinx-theme from pip to conda... and then I grouped it with other sphinx things.

File: docs/source/conf.py [hunk 2]
```diff
@@ -181,6 +181,6 @@
 
 # -- Extension configuration -------------------------------------------------
 intersphinx_mapping = {
-    'python': ('https://docs.python.org/3/', None),
-    'mdanalysis': ('https://docs.mdanalysis.org/stable/', None),
+    "python": ('https://docs.python.org/3/', None),
+    "mdanalysis": ('https://docs.mdanalysis.org/stable/', None),
```
Comment (orbeckst) [inline docs/source/conf.py:185]: same here... doubled up on work that was already in main

**Comment by lilyminium** [comment]
@orbeckst I could reproduce the footnote issue but not the missing `.data.files` -- does that still occur to you for main?

The footnotes issue was because it was in the first sentence of the module docstring, which was then reproduced in the autosummary table but without a corresponding local bibliography to link to.

**Review by orbeckst — APPROVED** [review]
I generally agree with @lilyminium that a global bibliography makes a lot of sense for this kit. However, the current PR is strictly improvement over the previous approach so let's merge it.

There's the one issue with missing(??) citation in the autosummary generated page
```
~/MDAnalysis/mdaencore/docs/source/autosummary/mdaencore.rst:38:<autosummary>:1: ERROR: Too many autonumbered footnote references: only 0 corresponding footnotes available.
~/MDAnalysis/mdaencore/docs/source/autosummary/mdaencore.rst:38:<autosummary>:1: ERROR: Unknown target name: "footcite-lindorfflarsen2009".
```
which I don't understand (and I am not a fan of autosummary in general...).

**Comment by orbeckst** [comment]
Thanks for fixing the footnote bib!

**Comment by orbeckst** [comment]
data files is still missing for me
```
...
building [html]: targets for 21 source files that are out of date
updating environment: [new config] 21 added, 0 changed, 0 removed
reading sources... [100%] index
WARNING: autodoc: failed to import module 'files' from module 'mdaencore.data'; the following exception was raised:
No module named 'mdaencore.data.files'
looking for now-outdated files... none found
pickling environment... done
checking consistency... ~/MDAnalysis/mdaencore/docs/source/autosummary/mdaencore.data.rst: WARNING: document isn't included in any toctree
~/MDAnalysis/mdaencore/docs/source/autosummary/mdaencore.data.files.rst: WARNING: document isn't included in any toctree
done
...
```

**Comment by orbeckst** [comment]
If RTD build then please merge whenever you want to @lilyminium .

**Comment by lilyminium** [comment]
Ah you might need to delete the autosummary directory and rebuild docs to get rid of the warning -- `mdaencore.data.files` was removed as a path, so it wouldn't be included in toctrees anymore, but if you built docs prior to its removal then its `rst` file would still be lying around.

**Comment by orbeckst** [comment]
I see. I assumed (wrongly) that `make clean` would be sufficient. But I also had to `rm -rf  source/autosummary/`. Now it worked without issues. Thanks for the tip.

(See, I am justified in not liking autosummary ;-) ).

**Comment by orbeckst** [comment]
The `setup-miniconda` job step timed out for two runners. Maybe change to `uses: mamba-org/setup-micromamba@v1` (although that also requires some option changing)... issue for another day.

**Pull request merged**